test/model/message_test.dart

```diff
@@ -375,6 +375,40 @@ void main() {
     });
   });
 
+  group('handleDeleteMessageEvent', () {
```

Comment:
This will also need tests for the behavior of `MessageListView`, as well as that of `MessageStoreImpl`. After all, the former is where most of the code this change introduces is. :slightly_smiling_face:

Those tests can go in `model/message_list_test.dart`; search for MessageEvent there for examples to follow. I think, given that we're using `_reprocessAll` rather than trying to do something more clever (as `handleMessageEvent` does), the only tests needed here are analogous to the tests with "MessageEvent" in their names.

Note one key thing those tests do is they call that `checkInvariants` function, in particular on every `notifyListeners`. It conducts a rather fierce set of consistency checks on the model, in particular recomputing everything that derives from `messages` and confirming it all has the right values.

Reply:
> Note one key thing those tests do is they call that checkInvariants function, in particular on every notifyListeners. It conducts a rather fierce set of consistency checks on the model, in particular recomputing everything that derives from messages and confirming it all has the right values.

I see that in the 'MessageEvent, before fetch' test, but not the other 'MessageEvent' tests. Was that intended?

Reply:
It's there :slightly_smiling_face: Via this part:
> in particular on every `notifyListeners`.

See `prepare` for that call site.

Reply:
(Quite possibly we should have `checkNotNotified` call `checkInvariants` too, and then the individual test cases will be more uniform in not needing to explicitly do so)

Reply:
Oh, I see! So the call is added in a listener. Then the post event check of 'MessageEvent, not in narrow' seems to be missing because it isn't expecting a notification. Yeah it makes sense to add it to `checkNotNotified` too.